meetings/working-groups/collection-literals/meetings/working-groups/collection-literals/CL-2023-06-16.md

```diff
@@ -0,0 +1,125 @@
+# Collection literals working group meeting for June 26, 2023
+
+## Agenda
+
+* Collection expression lifetimes
+* BCL apis
+* Breaking changes
+* Extension method lookup
+* Syntactic disambiguation
+
+## Discussion
+
+### Collection expression lifetimes
+
+Relevant link: https://gist.github.com/jaredpar/a784f8963ce73613a0a696ef89a00d9a
+
+Discussed the rules about spans produced by collection literals, and whether those are 'safe to return' from the 'current method'.
+
+Conclusion.  We will accept the rules in the provided link. Specifically, spans from collection literals are *not* "Safe to return from current method". This has the benefit of allowing the compiler to potentially stackalloc these spans for very high efficiency code (including cases like `G([1, 2, 3])`).
+
+While the compiler is free to use stackallocs for these spans, it is not required to do so.  However, as some consumers *may* only want to allow collection literals when they will be stackalloc'ed we will ensure that the compiler emits a hidden-diagnostic if it heap allocs.  These consumers can set that diagnostic to be an error, blocking unwanted span heap allocations.  In those cases, they can manually stackalloc.
+
+Users who want to be able to return those spans can do so by explicit conversion to an array.  e.g. `Span<int> s = (int[])[1, 2, 3];`. 
```

Comment:
I think at some point Chuck and I will work together to make the relevant revisions to the collection literals spec.